asyncio_mqtt/client.py

```diff
@@ -416,14 +426,30 @@ async def connect(self, *, timeout: int = 10) -> None:
         except (OSError, mqtt.WebsocketConnectionError) as error:
             raise MqttError(str(error)) from None
         await self._wait_for(self._connected, timeout=timeout)
+        # If _disconnected is already completed after connecting, reset it.
+        if self._disconnected.done():
+            self._disconnected: asyncio.Future[int | mqtt.ReasonCodes | None] = asyncio.Future()  # type: ignore[no-redef]
```

Comment:
Skip the type annotation here. We only need it in `__init__` during the member variable declaration. 👍